gossip/client_test.go

```diff
@@ -138,26 +138,32 @@ func TestClientGossip(t *testing.T) {
 		}
 	}()
 
-	if err := local.AddInfo("local-key", nil, time.Second); err != nil {
+	if err := local.AddInfo("local-key", nil, time.Hour); err != nil {
 		t.Fatal(err)
 	}
-	if err := remote.AddInfo("remote-key", nil, time.Second); err != nil {
+	if err := remote.AddInfo("remote-key", nil, time.Hour); err != nil {
 		t.Fatal(err)
 	}
 
 	// Use an insecure context. We're talking to tcp socket which are not in the certs.
 	rpcContext := rpc.NewContext(&base.Context{Insecure: true}, nil, stopper)
-	client.start(local, disconnected, rpcContext, stopper)
+	disconnected <- client
 
-	util.SucceedsSoon(t, func() error {
```

Comment:
IMO, using `SucceedsSoon` like this is an anti-pattern as it introduces a lower test timeout than the one specified on the command line. The lower timeout does not speed up the test in the common case, but makes it more flaky if the test is being run on a slow machine. Better to use the command line timeout. You're correct that changing this isn't necessary, but I think it is worthwhile.